src/components/Menu/Menu.jsx

```diff
@@ -0,0 +1,147 @@
+import React, {useCallback, useState, useEffect} from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'clsx';
+import styles from './Menu.scss';
+
+export const Menu = ({children, isDisplayed, minWidth, maxWidth, className, onMouseEnter, onMouseLeave, anchorEl, anchorPosition, onClose, ...props}) => {
+    const [stylePosition, setStylePosition] = useState();
+
+    useEffect(() => {
+        if (isDisplayed) {
+            setPositioningStyles(anchorEl);
+        }
+    }, [anchorEl, isDisplayed, setPositioningStyles]);
+
+    const setPositioningStyles = useCallback(() => {
+        const positioning = getPositioningStyle(anchorEl);
+        const positioningStyles = {
+            top: `calc(${positioning.top}px + ${anchorPosition.top})`,
+            left: `calc(${positioning.left}px + ${anchorPosition.left})`
+        };
+
+        setStylePosition(positioningStyles);
+    }, [getPositioningStyle, anchorEl, anchorPosition]);
```

Comment:
I'm not very familiar with useCallback, but what does it mean to have another callback as a dependency ? Shouldn't we use a simple useMemo ( `const positionStyle = useMemo(() => ...), [anchorEl])` , then `setPositioningStyles = useCallback(()=> {...}, [positionStyle, anchorPosition]` ? Or even more simple, if getPositioningStyle() is only used in setPositioningStyles(), just use one single useCallback().

Reply:
I have to confess this part of code is mostly inspired by the component popover from materia-ui.